dexlib2/src/main/java/org/jf/dexlib2/dexbacked/DexBackedField.java

```diff
@@ -131,4 +139,43 @@ private int getFieldIdItemOffset() {
         }
         return fieldIdItemOffset;
     }
+
+    /**
+     * Calculate and return the private size of a field definition.
+     *
+     * Calculated as: field_idx_diff + access_flags + annotations overhead +
+     * initial value size + field reference size
+     *
+     * @return size in bytes
+     */
+    public int getSize() {
+        int size = 0;
+        DexReader reader = dexFile.readerAt(startOffset);
+        reader.readLargeUleb128(); //field_idx_diff
+        reader.readSmallUleb128(); //access_flags
+        size += reader.getOffset() - startOffset;
+
+        Set<? extends DexBackedAnnotation> annotations = getAnnotations();
+        if (!annotations.isEmpty()){
+            size += 2 * 4; //2 * uint overhead from field_annotation
+            size += 4; //uint overhead for annotations_off
+            for (DexBackedAnnotation dexBackedAnnotation : annotations) {
```

Comment:
I believe annotation_set_item items can be shared, so I'm not sure if you want to include them here or not.